src/bootchart/store.c

```diff
@@ -409,6 +410,61 @@ int log_sample(DIR *proc,
                 ps->total = (ps->last->runtime - ps->first->runtime)
                             / 1000000000.0;
 
+                /* Take into account CPU runtime/waittime spent in non-main threads of the process
+                 * by parsing "/proc/[pid]/task/[tid]/schedstat" for all [tid] != [pid]
+                 * See https://github.com/systemd/systemd/issues/139
+                 */
+
+                /* Browse directory "/proc/[pid]/task" to know the thread ids of process [pid] */
+                snprintf(filename, sizeof(filename), PID_FMT "/task", pid);
+                taskfd = openat(procfd, filename, O_RDONLY|O_DIRECTORY|O_CLOEXEC);
+                if (taskfd != -1) {
+                        _cleanup_closedir_ DIR *taskdir = NULL;
+
+                        taskdir = fdopendir(taskfd);
+                        FOREACH_DIRENT(ent, taskdir, goto task_next) {
```

Comment:
Again, why not use `break` as the last argument here? You'd need to remove the `_cleanup_close_` attribute from `int taskfd`, which isn't needed anyway AFAICS …